kafka-connect-s3/src/test/java/io/confluent/connect/s3/DataWriterAvroTest.java

```diff
@@ -165,6 +165,29 @@ public void testWriteRecordsOfEnumsWithEnhancedAvroData() throws Exception {
     verify(sinkRecords, validOffsets);
   }
 
+  @Test
+  public void testNullValue() throws Exception {
+    localProps.put(StorageSinkConnectorConfig.ENHANCED_AVRO_SCHEMA_SUPPORT_CONFIG, "true");
+    localProps.put(StorageSinkConnectorConfig.CONNECT_META_DATA_CONFIG, "true");
+    localProps.put(S3SinkConnectorConfig.BEHAVIOR_ON_NULL_VALUES_CONFIG, "ignore");
+    setUp();
+    task = new S3SinkTask(connectorConfig, context, storage, partitioner, format, SYSTEM_TIME);
+
+    TopicPartition tp = context.assignment().iterator().next();
+    List<SinkRecord> sinkRecords = Collections
+        .singletonList(new SinkRecord(TOPIC, tp.partition(), null, "key", null, null, 42));
+    task.put(sinkRecords);
+    task.getTopicPartitionWriter(tp).commitFiles();
```

Comment:
this hasn't been required so far in tests. We kept forcing commits by hitting the corresponding config (e.g. write 3 records with flush.size equal to 3). I suggest basing the test on previous ones without the requirement to change the scope to the method above.

Reply:
I think explicitly flushing greatly improves test legibility, and separates concerns out, lets this test focus on what it focuses on. I think the behavior of the other tests is very unobvious

Reply:
To illustrate the point, when I originally wrote the test, I saw no output files, which I thought might be expected behavior, then after going through a different case much later, I realized i should be seeing an empty file, not a missing file. Reading the test, it should be obvious what is happening, but a mystery flush size makes it quite challenging, and forces all the unit tests to conform or pass a ton of args around

Reply:
I'm confused why you refer to flush a mystery. There are lots of tests exercising what `put` does, including timebased and size based rotations. You assert the result in several ways using the `verifyOffsets` and `verify` methods. I don't see why we can't follow the same pattern here